test/desdemona/query_test.clj

```diff
@@ -29,31 +29,62 @@
   (is (= '(= (:ip x) "10.0.0.1")
          (q/infix->dsl "ip(x) = 10.0.0.1"))))
 
-(def dsl->logic
-  @#'desdemona.query/dsl->logic)
+(deftest free-sym-tests
+  (is (not (#'q/free-sym? 's))
+      "sym not marked as free")
+  (is (not (#'q/free-sym? 1))
+      "not a symbol")
+  (is (#'q/free-sym? (#'q/free-sym 's))
+      "sym explicitly marked as free"))
+
+(deftest find-free-vars-tests
+  (are [expected query] (= expected (#'q/find-free-vars query))
+    #{}
+    '()
+
+    #{'x}
+    (#'q/dsl->logic '(= (:ip x) "10.0.0.1"))
+
+    #{'y}
+    (#'q/dsl->logic '(= (:ip y) "10.0.0.1"))
```

Comment:
Correct; its only job is to demonstrate that `dsl->logic` works when you use `y` instead of `x`.